qiskit/optimization/algorithms/grover_optimizer.py

```diff
@@ -0,0 +1,379 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""GroverOptimizer module"""
+
+import logging
+from typing import Optional, Dict, Union, Tuple
+import math
+import random
+import numpy as np
+from qiskit.aqua import QuantumInstance
+from qiskit.optimization import QiskitOptimizationError
+from qiskit.optimization.algorithms import OptimizationAlgorithm
+from qiskit.optimization.problems import QuadraticProgram
+from qiskit.optimization.converters import (QuadraticProgramToQubo,
+                                            QuadraticProgramToNegativeValueOracle)
+from qiskit.optimization.algorithms.optimization_algorithm import OptimizationResult
+from qiskit.aqua.algorithms.amplitude_amplifiers.grover import Grover
+from qiskit import Aer, QuantumCircuit
```

Comment:
Can we remove defaulting of the quantum instance. Especially to Aer; if it really must be there then use BasicAer which is part of Terra rather than Aer

Reply:
I think this is fair, let's remove it, this is something the user should consciously set. However I'd prefer to keep it optional, add a setter and check that is has been set before solving. This way we don't force setting it in the beginning.